### ISSUE TITLE
Small problem with DRF js client

### ISSUE THREAD

**Issue by jojolalpin** [evenicoulddoit/django-rest-framework-serializer-extensions#9]
Hi,

First, thanks for your package.
When using the js client from django DRF, it calls the get_serializer_context() method without any request.
So in views.py, SerializerExtensionsAPIViewMixin class, method get_extensions_mixin_context (line 41), request is None and method fails.
```
if params_enabled:
    query_params = self.request.query_params.getlist(field) 
```
Adding a simple condition solves this.
```
if params_enabled and self.request is not None:
    query_params = self.request.query_params.getlist(field) 
```

**Comment by evenicoulddoit** [comment]
Hey @jojolalpin - thanks for this and the PR accompanying. I've not used any of the DRF API Clients before, so you might have to help me out a little.

From the docs it looks like the client is used to fire of XMLHttpRequests, is that correct? Seems weird that these wouldn't have a request - but that aside - are you trying to leverage extensions when using your JS client? How would it pass the params if not through the request? From your PR it looks like this essentially disables the params in the case that no request exists. Just trying to get a handle for the use-case.

Thanks!

**Comment by jojolalpin** [comment]
Hi Ian,

To use the js DRF client, you request the js library at ...docs/schema.js
but while generating it with get_schema method of file
rest_framework/schemas.py, request is lost in get_links (line 342).
With the modification, params are disabled only if there is no request,
what would generate an error in all cases and should almost only happen in
very special cases.
I think it is still correct as no request implies no params.

Best regards,

**Comment by evenicoulddoit** [comment]
Ah - I see ([assuming you're referring to this](https://github.com/encode/django-rest-framework/blob/871ce3498380299e742cbb38ee810c4b02e4d0c0/rest_framework/schemas.py#L318)) - so this is used as part of the API discovery process. Interesting that it calls `get_serializer_context()` to achieve that...

So - to immediately fix the discover I can use your solution, but I wonder if I should also look into how the clients work at a later state to see if I can hook up knowledge about the expandable fields?

**Comment by jojolalpin** [comment]
Sure you may add value to the client.
For what I have read, client is a new feature and functionalities should
improve in a near future.
I did not investigate how it works.
As for now, I think this quick fix restricts nothing.

**Comment by evenicoulddoit** [comment]
@jojolalpin I merged #11 yesterday, which should do the job - I'll make a new build in the next few days. Thanks!

**Comment by jojolalpin** [comment]
Thank you very much Ian.

**Comment by jojolalpin** [comment]
Hi Ian,
When do you plan to publish a new pypi package ?
It is to know if I install from the git or wait for the new version.
best regards,

**Comment by evenicoulddoit** [comment]
Hi @jojolalpin - thanks for the nudge, should have published the `0.5.2` release now

**Comment by jojolalpin** [comment]
Thanks a lot !
